web/src/pages/setting/SettingContent.vue

```diff
@@ -0,0 +1,98 @@
+<script lang="ts" setup>
+export interface LanguageWithIndex {
+  index: number
+}
+export interface LanguageWithIndex extends Language { }
+//LanguageWithIndex interface is merged by the Language type and custome index prop
+
+import { Language, languages } from "@/services/language"
+import {
+  Dialog,
+  DialogPanel,
+  DialogTitle,
+  DialogDescription,
+  TransitionChild,
+  TransitionRoot,
+  TabGroup,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
+} from "@headlessui/vue"
+import LanguageSelector from "@/components/settings/LangSelector.vue"
+import { ref } from "vue"
+import { useI18n } from "vue-i18n"
+const { t } = useI18n()
+
+interface props {
```

Comment:
Keep to the default type name convention (PascalCase).

In this case you probably don't need to define a separate interface, just put it directly into the type parameters on defineProps